src/main/java/org/fogbowcloud/manager/core/plugins/identity/openstack/KeystoneV3IdentityPlugin.java

```diff
@@ -0,0 +1,277 @@
+			Integer statusResponse = HttpStatus.SC_BAD_REQUEST;
+			throw new RuntimeException(statusResponse.toString());
+		} catch (Exception e) {
+			LOGGER.error("Could not do post request.", e);
+			Integer statusResponse = HttpStatus.SC_BAD_REQUEST;
+			throw new RuntimeException(statusResponse.toString());
+		} finally {
+			try {
+				EntityUtils.consume(response.getEntity());
+			} catch (Exception e) {
+				LOGGER.error("Could not release HTTPEntity resources.", e);
+			}
+		}
+		checkStatusResponse(response);
+
+		return new Response(responseStr, response.getAllHeaders());
+	}
+
+	private HttpClient getClient() {
+		return client;
+	}
+
+	protected JSONObject mountJson(Map<String, String> credentials) throws JSONException {
+		JSONObject projectId = new JSONObject();
+		projectId.put(ID_PROP, credentials.get(PROJECT_ID));
+		JSONObject project = new JSONObject();
+		project.put(PROJECT_PROP, projectId);
+
+		JSONObject userProperties = new JSONObject();
+		userProperties.put(PASSWORD_PROP, credentials.get(PASSWORD));
+		userProperties.put(ID_PROP, credentials.get(USER_ID));
+		JSONObject password = new JSONObject();
+		password.put(USER_PROP, userProperties);
+
+		JSONObject identity = new JSONObject();
+		identity.put(METHODS_PROP, new JSONArray(new String[] { PASSWORD_PROP }));
+		identity.put(PASSWORD_PROP, password);
+
+		JSONObject auth = new JSONObject();
+		auth.put(SCOPE_PROP, project);
+		auth.put(IDENTITY_PROP, identity);
+
+		JSONObject root = new JSONObject();
+		root.put(AUTH_PROP, auth);
+		
+		return root;
+	}
+
+	private Token getTokenFromJson(Response response) {
+		
+		String accessId = null;
+		Header[] headers = response.getHeaders();
+		for (Header header : headers) {
+			if (header.getName().equals(X_SUBJECT_TOKEN)) {
+				accessId = header.getValue();
+			}
+		}
+		
+		try {
+			JSONObject root = new JSONObject(response.getContent());
+			JSONObject token = root.getJSONObject(TOKEN_PROP);
+
+			JSONObject user = token.getJSONObject(USER_PROP);
+			String userId = user.getString(ID_PROP);
+			String userName = user.getString(NAME_PROP);
+
+			Map<String, String> tokenAtt = new HashMap<String, String>();
+			String tenantId = null;
+			String tenantName = null;
+			try {
+				tenantId = token.getJSONObject(PROJECT_PROP).getString(ID_PROP);
+				tokenAtt.put(TENANT_ID, tenantId);
+			} catch (JSONException e) {
+				LOGGER.debug("There is no tenantId inside json response.");
+			}
+			try {
+				tenantName = token.getJSONObject(PROJECT_PROP).getString(NAME_PROP);
+				tokenAtt.put(TENANT_NAME, tenantName);
+			} catch (JSONException e) {
+				LOGGER.debug("There is no tenantName inside json response.");
+			}
+
+			return new Token(accessId, new Token.User(userId, userName), new Date(), tokenAtt);
+		} catch (Exception e) {
+			LOGGER.error("Exception while getting token from json.", e);
+			return null;
+		}
+	}
+
+	private void checkStatusResponse(HttpResponse response) {
```

Comment:
I saw a same method in the @llvieira [PR](https://github.com/fogbow/fogbow-manager-core/pull/21), could you and @llvieira discuss about a solution for this code duplication? I suspect that this checkStatusResponse will be used by others classes that handles HTTP requests :smile: